src/Database/index.js

```diff
@@ -88,4 +93,8 @@ export default class Database {
       collection.unsafeClearCache()
     })
   }
+
+  _ensureInAction(error: string): void {
```

Comment:
Hmm is this safe? I mean there might be other actions in queue and I could use `Collection.create` outside of an action and this wouldn't throw an error. Am I right? :P Sometimes it's kinda hard to follow the flow in async code :/

Reply:
You're right! Asynchronicity is really hard! This solution does not give you 100% guarantee of correctness, but you should be able to catch 99% errors, because you'd need to constantly have a long-running action to not hit the error.

I think the only way to guarantee correctness would be to remove batch/create/update methods, and have those methods only on an object passed to the `database.action()` block

Reply:
Ok, thanks for explanation :) I guess 99% is good enough for now :)